### PR TITLE
Add article "Porting a Dancer plugin to Dancer2"

### PR DESCRIPTION
**Pull request by cxw42**
As promised in https://github.com/tpf/perldotcom/issues/215#issuecomment-568783103 .  Thanks for considering this submission!

### CODE REVIEW

**Comment by dnmfarrell** [comment]
Thanks for this Chris, I'll take a look and let you know.

**Comment by briandfoy** [comment]
I like it. Aside from any light editing, I think it's good to go.

**Comment by cxw42** [comment]
@briandfoy Much appreciated!

File: content/article/porting-a-dancer-plugin-to-dancer2.md [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+    "draft"       : true,
+    "image"       : "",
+    "thumbnail"   : "",
+    "description" : "How I ported a plugin without having to know it all first",
+    "categories"  : "development"
+  }
+
+----------
+
+In my [Dancer2]({{< mcpan "Dancer2" >}}) web application, I want to know which requests come from smartphones.  There’s a plugin for that — but only in the older [Dancer (v1)]({{< mcpan "Dancer" >}}) framework.  I’m no expert, but even I was easily able to port the Dancer plugin, [Dancer::Plugin::MobileDevice]({{< mcpan "Dancer::Plugin::MobileDevice" >}}), to Dancer2!  In this article, we’ll explore Dancer2 and the way it handles plugins.  We’ll get our hands dirty working with the with the framework, and examine the main changes I made to port the plugin from Dancer to Dancer2.  By the end of this article, you’ll be ready to rock _and_ you’ll have a handy reference to use when porting plugins yourself.
```
Comment (dnmfarrell) [inline content/article/porting-a-dancer-plugin-to-dancer2.md]: "with the" repeats here

File: content/article/porting-a-dancer-plugin-to-dancer2.md [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+    "date"        : "2020-01-10T13:37:00",
+    "tags"        : ['dancer','dancer2','perl','perl-5','perl-programming','web-development','porting','plugin'],
+    "draft"       : true,
+    "image"       : "",
+    "thumbnail"   : "",
+    "description" : "How I ported a plugin without having to know it all first",
+    "categories"  : "development"
+  }
+
+----------
```
Comment (dnmfarrell) [inline content/article/porting-a-dancer-plugin-to-dancer2.md]: We usually don't use line breaks here - if you're set on having one it's fine by me though.

File: content/article/porting-a-dancer-plugin-to-dancer2.md [hunk 3]
```diff
@@ -0,0 +1,177 @@
+  {
+    "title"       : "Porting a Dancer plugin to Dancer2",
+    "authors"     : ["christopher-white"],
+    "date"        : "2020-01-10T13:37:00",
+    "tags"        : ['dancer','dancer2','perl','perl-5','perl-programming','web-development','porting','plugin'],
```
Comment (dnmfarrell) [inline content/article/porting-a-dancer-plugin-to-dancer2.md]: `s/'/"/g`

File: content/article/porting-a-dancer-plugin-to-dancer2.md (hunk 2 above)
Comment (cxw42) [reply content/article/porting-a-dancer-plugin-to-dancer2.md]: Based on your recent https://github.com/tpf/perldotcom/commit/7e498d266efc9cbbabdbfb5fba3a4975f92d405a , I think you mean line 13 should be deleted.  Am I understanding you correctly?  If so, no problem!

Comment (dnmfarrell) [reply content/article/porting-a-dancer-plugin-to-dancer2.md]: Yes, thanks :+1:

**Comment by cxw42** [comment]
@dnmfarrell Updated - should be all set!  Thanks!

**Pull request merged**